// frontend/assets/api_config.js
// API Configuration

/**
 * Offline WebView configuration: BASE_URL ignored, local assets used
 */
const API_CONFIG = {
  BASE_URL: 'offline',
  TIMEOUT: 10000,
  RETRY_ATTEMPTS: 3,
  GOOGLE_CLIENT_ID: '455059237598-3m7hau4maupqe57gbpa3tkk8t512m541.apps.googleusercontent.com',
  MAPS_API_KEY: 'AIzaSyB8VoJZmeh79TYEFM4wcUUdOsU-ltz_UM0'
};

const BASE_ASSET_PATH = 'assets/';
window.BASE_ASSET_PATH = BASE_ASSET_PATH;

async function fetchLocalJSON(path) {
  try { const res = await fetch(path); if (!res.ok) return null; return await res.json(); } catch (_) { return null; }
}

async function apiCall(endpoint, options = {}) {
  const e = String(endpoint || '').toLowerCase();
  if (e.includes('remedies')) return await getOfflineRemedies();
  if (e.includes('first-aid') || e.includes('firstaid')) return await getOfflineFirstAid();
  if (e.includes('symptoms')) return await getOfflineSymptoms();
  if (e.includes('quiz')) return await getOfflineQuiz();
  return { success: false, message: 'Offline mode: endpoint not available', endpoint };
}

async function getOfflineRemedies() {
  try {
    const d = localStorage.getItem('remediesData');
    if (d) return { success:true, data: JSON.parse(d) };
    const data = await fetchLocalJSON(BASE_ASSET_PATH + 'remedies_data.json');
    if (data) { const arr = Array.isArray(data) ? data : (data.data || data.items || []); localStorage.setItem('remediesData', JSON.stringify(arr)); return { success:true, data: arr }; }
  } catch (_) {}
  return { success:false, message:'No offline data available' };
}

async function getOfflineFirstAid() {
  try {
    const d = localStorage.getItem('firstAidData');
    if (d) return { success:true, data: JSON.parse(d) };
    const data = await fetchLocalJSON(BASE_ASSET_PATH + 'first_aid_data.json');
    if (data) { const arr = Array.isArray(data) ? data : (data.data || data.items || []); localStorage.setItem('firstAidData', JSON.stringify(arr)); return { success:true, data: arr }; }
  } catch (_) {}
  return { success:false, message:'No offline data available' };
}

async function getOfflineSymptoms() {
  try {
    const d = localStorage.getItem('symptomsData');
    if (d) return { success:true, data: JSON.parse(d) };
    const data = await fetchLocalJSON(BASE_ASSET_PATH + 'symptoms_data.json');
    if (data) { const arr = Array.isArray(data) ? data : (data.data || data.items || []); localStorage.setItem('symptomsData', JSON.stringify(arr)); return { success:true, data: arr }; }
  } catch (_) {}
  return { success:false, message:'No offline data available' };
}

async function getOfflineQuiz() {
  try {
    const d = localStorage.getItem('quizQuestions');
    if (d) return { success:true, data: JSON.parse(d) };
    const data = await fetchLocalJSON(BASE_ASSET_PATH + 'quiz_first_aid.json');
    if (data) { const arr = Array.isArray(data) ? data : (data.data || data.items || []); localStorage.setItem('quizQuestions', JSON.stringify(arr)); return { success:true, data: arr }; }
  } catch (_) {}
  return { success:false, message:'No offline data available' };
}

if (typeof module !== 'undefined' && module.exports) {
  module.exports = { API_CONFIG, apiCall, getMapsApiKey, setMapsApiKey };
} else {
  window.API_CONFIG = API_CONFIG;
  window.apiCall = apiCall;
  window.getMapsApiKey = getMapsApiKey;
  window.setMapsApiKey = setMapsApiKey;
}

function getMapsApiKey() { try { const ls = localStorage.getItem('GOOGLE_MAPS_API_KEY'); if (ls && ls.trim()) return ls.trim(); } catch(e) {} return (API_CONFIG && API_CONFIG.MAPS_API_KEY) ? API_CONFIG.MAPS_API_KEY : ''; }
function setMapsApiKey(k) { try { if (typeof k === 'string' && k.trim()) { localStorage.setItem('GOOGLE_MAPS_API_KEY', k.trim()); return true; } } catch(e) {} return false; }
